src/change_stream/event.rs

```diff
@@ -15,7 +23,31 @@ use serde::{Deserialize, Serialize};
 /// [here](https://docs.mongodb.com/manual/changeStreams/#change-stream-resume-token) for more
 /// information on resume tokens.
 #[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct ResumeToken(pub(crate) Bson);
+pub struct ResumeToken(pub(crate) RawBson);
+
+impl ResumeToken {
+    pub(crate) fn initial(
```

Comment:
nit: can we update this so that both values don't always need to be created? e.g.
```
match spec.post_batch_resume_token {
    Some(token) if spec.initial_buffer.is_empty() => token,
    None => // token from options
}
```

Reply:
That's much nicer, thank you!